Source/WTF/wtf/StackBounds.cpp

```diff
@@ -158,7 +158,19 @@ StackBounds StackBounds::currentThreadStackBoundsInternal()
 WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN
         void* bound = static_cast<char*>(origin) - size;
 WTF_ALLOW_UNSAFE_BUFFER_USAGE_END
-        return StackBounds { origin, bound };
+
+        static char** oldestEnviron = environ;
```

Comment:
maybe I'm misunderstanding, shouldn't this be youngestEnviron? As, whenever this is called we set it to the latest environ?

Reply:
Ah nvm I misread, it is initialized only once, hence oldest.